homeassistant/components/mqtt/fan.py

```diff
@@ -473,6 +506,33 @@ def oscillation_received(msg: ReceiveMessage) -> None:
             }
             self._attr_oscillating = False
 
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def direction_received(msg: ReceiveMessage) -> None:
+            """Handle new received MQTT message for the directionoscillation."""
+            payload = self._value_templates[ATTR_DIRECTION](msg.payload)
+            if not payload:
+                _LOGGER.debug("Ignoring empty direction from '%s'", msg.topic)
+                return
+            if payload == self._payload["OSCILLATE_DIRECTION_FORWARD"]:
+                self._attr_current_direction = self._payload[
+                    "OSCILLATE_DIRECTION_FORWARD"
+                ]
+            if payload == self._payload["OSCILLATE_DIRECTION_REVERSE"]:
+                self._attr_current_direction = self._payload[
+                    "OSCILLATE_DIRECTION_REVERSE"
+                ]
```

Comment:
It seems you do not refer to the payload from the config schema here, further you should assign `DIRECTION_FORWARD`, `DIRECTION_REVERSE` or `None` when assigning `self._attr_current_direction`

Reply:
True, should be fixed now